### ISSUE TITLE
Material changes within objects are not handled

### ISSUE THREAD

**Issue by jansol** [Twinklebear/tobj#9]
Should probably emit a separate model on every material change.

**Comment by Twinklebear** [comment]
I think instead of a separate model on each change we should track per-triangle material indices and emit these (similar to [tinyobjloader](https://github.com/syoyo/tinyobjloader/)). Though I'm not sure which would be easier to use, per-face vs. new object on each material change.

If we do per-triangle materials the user then has to batch the models up for rendering themselves based on batching the material IDs, but if we emit a new model for each material change, we may actually emit more models than we need? For example, if the model has 4 `usemtl` as we parse though and changes material 4 times but only between 2 materials we'd emit 4 models when 2 would be what the user would batch up using the per-triangle materials. However, I'm not sure if OBJ exporters would output the file like that? It seems like it would be a pretty poor output from the OBJ exporter, so maybe we wouldn't actually run into that situation? Tinyobjloader does per-face material output, so maybe it does come up? I'm not sure.

**Comment by jansol** [comment]
I just found out that ticking the "Material Groups" box in blender's obj exporter solves this for me by emitting explicit groups on every material change which solves this for my case.

Generally speaking though, I'm not sure whether per-triangle materials or consolidating all chunks of one material in the loader is better. In most cases (when only caring about rendering) it'll be the latter, but I guess in some editor-like use case one might want to keep the order intact etc. Also obj files containing actually separate objects would probably suffer if such behaviour were implemented.
